### PR TITLE
Vcs/electrical profile endpoint

### LINKED ISSUE #2891: Add an endpoint to post an electrical profile
## Requirements

The pr https://github.com/DGEXSolutions/osrd/pull/2886 needs to be merged.

## Acceptance Criteria

- [ ] Have an editoast command  that add a set of electrical profiles `editoast add-profile-set custom_france path/electrical_profile.json`
- [ ] Add a `POST` endpoint that add an electical profile.
  - curl --location --request POST 'http://localhost:8090/electrical_profile_set/?name=Toto' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--data-raw '{...}'

## Implementation help

- You need to add pg tables if not existing
- Both endpoint and command should call the same code
- Use existing ElectricalProfile schema

### PR DESCRIPTION
**Pull request by anisometropie**
closes #2891

### CODE REVIEW

File: editoast/src/client/mod.rs [hunk 1]
```diff
@@ -24,6 +24,7 @@ pub enum Commands {
     Generate(GenerateArgs),
     Clear(ClearArgs),
     ImportRailjson(ImportRailjsonArgs),
+    AddProfileSet(AddProfileSetArgs),
```
Comment (flomonster) [inline editoast/src/client/mod.rs]: We can rename it for consistency.

```suggestion
    ImportProfileSet(ImportProfileSetArgs),
```

File: editoast/src/views/electrical_profiles.rs [hunk 2]
```diff
@@ -48,12 +49,31 @@ async fn get(
...
+        let mut conn = db_pool.get().expect("Failed to get DB connection");
+        let electrical_profile_set = ElectricalProfileSet::create_electrical_profile_set(
+            &mut conn,
+            name.into_inner(),
+            data.into_inner(),
+        )?;
+        Ok(Json(electrical_profile_set))
+    })
+    .await
+    .unwrap()
```
Comment (flomonster) [inline editoast/src/views/electrical_profiles.rs]: `ElectricalProfileSet::create_electrical_profile_set` should take a `Data<DbPool>` argument and be responsible to block and get a PgConnection.

File: editoast/src/views/electrical_profiles.rs [hunk 3]
```diff
@@ -71,14 +91,21 @@ pub struct ElectricalProfileSetMetaData {
...
 #[diesel(table_name = osrd_infra_electricalprofileset)]
 pub struct ElectricalProfileSet {
     pub id: i64,
     pub name: String,
     pub data: JsonValue,
 }
 
+#[derive(Debug, PartialEq, Insertable, Queryable)]
+#[diesel(table_name = osrd_infra_electricalprofileset)]
+pub struct NewElectricalProfileSet {
```
Comment (flomonster) [inline editoast/src/views/electrical_profiles.rs]:
```suggestion
struct NewElectricalProfileSet {
```

**Review by flomonster — CHANGES_REQUESTED** [review]
Great PR! Only a few things to adapt.

File: editoast/src/client/mod.rs (hunk 1 above)
Comment (anisometropie) [reply editoast/src/client/mod.rs]: done

File: editoast/src/views/electrical_profiles.rs (hunk 2 above)
Comment (anisometropie) [reply editoast/src/views/electrical_profiles.rs]: Can I get the existing DbBool in `src/main.rs` ? or do I need to use `Pool::builder()` ?

Screenshot_20230307_151159

Comment (anisometropie) [reply editoast/src/views/electrical_profiles.rs]: By the way, is it Okay to use the `view` ElectricalProfileSet here in `main.rs` ? I had to make `create_electrical_profile_set` public

File: editoast/src/views/electrical_profiles.rs (hunk 3 above)
Comment (anisometropie) [reply editoast/src/views/electrical_profiles.rs]: done

File: editoast/src/views/electrical_profiles.rs (hunk 2 above)
Comment (flomonster) [reply editoast/src/views/electrical_profiles.rs]: You can do this to get a `Data<Pool>`:

```rust
let pool = Data::new(Pool::builder().max_size(1).build(manager).unwrap());
```

-----------------

It's okay to use the view in the main but ideally, you would separate the view from the operations. For example, there are two module documents.

- The first one contains the operations you can perform.
- The second one contains the view that simply calls the operations.

Note: It's fine to keep it that way.

**Review by flomonster — DISMISSED** [review]
Well done!

**Review by flomonster — APPROVED** [review]

**Pull request merged**